lib/plugins/aws/package/compile/functions.js

```diff
@@ -102,18 +103,16 @@ class AwsCompileFunctions {
       _.get(functionObject, 'package.artifact') ||
       _.get(this, 'serverless.service.package.artifact');
 
-    const regex = new RegExp('s3\\.amazonaws\\.com/(.+)/(.+)');
-    const match = artifactFilePath.match(regex);
-
-    if (!match) return;
+    const s3Object = parseS3URI(artifactFilePath);
+    if (!s3Object) return;
+    if (process.env.SLS_DEBUG) {
+      this.serverless.cli.log(`Downloading ${s3Object.Key} from bucket ${s3Object.Bucket}`);
+    }
     await new Promise((resolve, reject) => {
       const tmpDir = this.serverless.utils.getTmpDirPath();
-      const filePath = path.join(tmpDir, match[2]);
+      const filePath = path.join(tmpDir, path.basename(s3Object.Key));
```

Comment:
Is `path.basename` here correct? technically it'll convert `foo/bar.x` into `bar.x` and if I see correctly, we're not doing that in master

Reply:
Good question! Yes, `path.basename` is needed because otherwise the write stream will attempt to write to `$tmpDir/path/to/artifact.zip` and those nested folders (`path/to/`) doesnt exist in that temp directory causing the write stream to fail (as it doesnt create folders for you). So I `path.basename()` the `Key` to ensure that it saves the `artifact.zip` file directly into the `$tmpDir` which does exist.

Reply:
Ok, I reviewed again old code, and see that there's no change here (as in regex _greedy_ operator was used, so at `match[2]` we always had basename)

👍